### PR TITLE
Common - Allow custom event name for status effects

### PR DESCRIPTION
**Pull request by PabstMirror**
Alt to #10468

Allows custom event names for status effects
Switch to hashmap

### CODE REVIEW

File: addons/common/functions/fnc_statusEffect_sendEffects.sqf [hunk 1]
```diff
@@ -21,31 +21,34 @@ TRACE_2("params",_object,_effectName);
...
-                };
-            };
+
+private _effectVarName = format [QGVAR(effect_%1), _effectName];
+private _effectNumber = _object getVariable [_effectVarName, -1];
+
+//We only do anything if the effect has been defined at some point in the game for this unit
+TRACE_2("checking if event is nil",_effectName,_effectNumber);
+if (_effectNumber != -1) then {
+    (GVAR(statusEffects) get toLowerANSI _effectName) params ["_isGlobal", "_sendJIP", "_eventName"];
```
Comment (johnb432) [inline addons/common/functions/fnc_statusEffect_sendEffects.sqf:37]: Do we want to support non-ANSI effect names too or not?

File: addons/common/functions/fnc_statusEffect_sendEffects.sqf [hunk 2]
```diff
@@ -21,31 +21,34 @@ TRACE_2("params",_object,_effectName);
 
 if (isNull _object) exitWith {};
 
-{
-    if ((_effectName == "") || {_effectName == _x}) then {
-        private _effectVarName = format [QGVAR(effect_%1), _x];
-        private _effectNumber = _object getVariable [_effectVarName, -1];
+if (_effectName == "") exitWith { // recurse though all possible effects
```
Comment (rautamiekka) [inline addons/common/functions/fnc_statusEffect_sendEffects.sqf:24]:
```suggestion
if (_effectName == "") exitWith { // recurse through all possible effects
```

File: addons/common/functions/fnc_statusEffect_sendEffects.sqf (hunk 1 above)
Comment (rautamiekka) [reply addons/common/functions/fnc_statusEffect_sendEffects.sqf:37]: IMO non-ANSI effect names would be totally unreasonable.

**Review by rautamiekka — CHANGES_REQUESTED** [review]

File: addons/common/functions/fnc_statusEffect_sendEffects.sqf (hunk 2 above)
Comment (DartRuffian) [inline addons/common/functions/fnc_statusEffect_sendEffects.sqf:24]:
```suggestion
if (_effectName == "") exitWith { // Recurse through all possible effects
```

File: addons/common/functions/fnc_statusEffect_sendEffects.sqf [hunk 3]
```diff
@@ -21,31 +21,34 @@ TRACE_2("params",_object,_effectName);
...
-                default {
-                    TRACE_2("Sending Target Event",_object,_effectNumber);
-                    [_eventName, [_object, _effectNumber], _object] call CBA_fnc_targetEvent;
-                };
-            };
+
+private _effectVarName = format [QGVAR(effect_%1), _effectName];
+private _effectNumber = _object getVariable [_effectVarName, -1];
+
+//We only do anything if the effect has been defined at some point in the game for this unit
```
Comment (DartRuffian) [inline addons/common/functions/fnc_statusEffect_sendEffects.sqf:34]:
```suggestion
// We only do anything if the effect has been defined at some point in the game for this unit
```

File: addons/common/functions/fnc_statusEffect_sendEffects.sqf (hunk 1 above)
Comment (johnb432) [reply addons/common/functions/fnc_statusEffect_sendEffects.sqf:37]: It doesn't matter if it's reasonable or not, it matters if it's possible. Given that `addType` forces all names to be `toLowerANSI`, it's impossible and therefore doesn't matter.

**Review by johnb432 — APPROVED** [review]
I haven't tested, but logic seems good to me.

**Pull request merged**